src/main/java/bio/terra/flight/dataset/ingest/IngestSetupStep.java

```diff
@@ -55,6 +64,34 @@ public StepResult doStep(FlightContext context) {
         String sgName = FlightUtils.randomizeNameInfix(baseName, "_st_");
         IngestUtils.putStagingTableName(context, sgName);
 
+        IngestRequestModel ingestRequestModel = IngestUtils.getIngestRequestModel(context);
+        IngestRequestModel.StrategyEnum ingestStrategy = ingestRequestModel.getStrategy();
+
+        if (ingestStrategy == IngestRequestModel.StrategyEnum.UPSERT) {
+            List<Column> primaryKey = dataset
+                .getTableByName(targetTable.getName())
+                .get()
```

Comment:
If this is an `Optional<List<Column>>` then we should not always assume that we can `.get()`. If we can't get the target table by name, let's throw a `CorruptMetadataException`.